### PR TITLE
Get latest version of signature files

### PR DESCRIPTION
**Pull request by MancunianSam**
When the lambda starts from cold, it carries out the following steps.

* Checks to see if the /tmp directory contains container-* or DROID_Signature*
* If they exist, use those to create the Droid instance.
* If they don't exist:
  * Call http://www.nationalarchives.gov.uk/pronom/container-signature.xml and get the last modified date
  * Convert this to YYYYMMDD and use that for the container signature version
  * Call the SOAP endpoint at http://www.nationalarchives.gov.uk/pronom/service.asmx
  * This returns the latest Droid signature file version.
  * Download both these files from the TNA CDN as before.

I've done it this way because we should only be calling the nationalarchives url when the lambda cold starts, if we're running a lot of files at once, it won't get called repeatedly. When the lambda does cold start though, we'll get the latest version.

If this doesn't work or we end up hitting the TNA website too often, we'll have to remove this and look at setting up a cron job to update it periodically but we have enough jobs like that already so I wanted to try it this way.

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -5,8 +5,7 @@ s3 {
 }
 
 signatures {
-  droid = ${DROID_SIGNATURE_VERSION}
-  container = ${CONTAINER_SIGNATURE_VERSION}
   cdn = "https://cdn.nationalarchives.gov.uk/documents"
+  version = "https://www.nationalarchives.gov.uk"
```
Comment (TomJKing) [inline src/main/resources/application.conf:9]: Is `version` the best name for this parameter, bit confusing as its used as the national archives URL in the code.

File: src/test/scala/uk/gov/nationalarchives/fileformat/SignatureFilesTest.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          createResponse(200, "test")
+        )
+      val containerSignatureFile = new File("container-signature-1.xml")
+      val droidSignatureFile = new File("DROID_SignatureFile_V1")
+      val existingFiles = List(containerSignatureFile, droidSignatureFile)
+      new SignatureFiles(client, existingFiles).downloadSignatureFile(signatureType)
+      versionCaptor.getAllValues.size should equal(0)
+    }
+
+    "downloadSignatureFile" should s"not call the droid api if the files do not exist for the $signatureType" in {
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/fileformat/SignatureFilesTest.scala:42]: Should the title of this test be: `should "call the droid api if the ....`

**Review by TomJKing — COMMENTED** [review]
Looks good. Couple of minor comments. Also would you be able to update the `README` about how this works now. What you've written in the comment for the PR should be OK.

File: src/main/resources/application.conf (hunk 1 above)
Comment (MancunianSam) [reply src/main/resources/application.conf]: Good point, I think I originally had it as version in the code and it was confusing. I've changed it here.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**